File: exams/exam_001/assignment_002/solution.go
package assignment002

func Call() {
	solve("PrAsAd")
	solve("hEllO WoRlD")
}

func solve(str string) string {
	runes := []byte(str)
	for indx, r := range runes {
		if r >= 92 && r <= 122 { // is lower case
			runes[indx] = (r - 32)
		} else if r >= 65 && r <= 90 {
			runes[indx] = (r + 32)
		}
	}
	return string(runes)
}

File: exams/exam_001/assignment_002/solution.rb
class Solution
  def call
    solve("PrAsAd")
    solve("hEllO WoRlD")
  end

  def solve(str)
    chars = str.split('')
    data = chars.map do |c|
      ord = c.ord
      if ord >= 92 && ord <= 122 # is lower case
        (ord - 32).chr
      elsif ord >= 65 && ord <= 90
        (ord + 32).chr
      else
        c
      end
    end
    data.join('')
  end
end

# puts '-' * 100
# puts Solution.new.solve("PrAsaD")
# puts '-' * 100
# puts Solution.new.solve("hEllO WoRlD")
